### PR TITLE
kmp refactor of C4-166 to use utils functionality

### PR DESCRIPTION
**Pull request by netsettler**
This takes a lot of the stuff from the underlying c4_166 branch and assumes it will be in utils. For the time being, it refers to a beta that's on a branch.

### CODE REVIEW

File: src/encoded/commands/create_mapping_on_deploy.py [hunk 1]
```diff
@@ -5,8 +5,10 @@
 from pyramid.paster import get_app
 from snovault.elasticsearch.create_mapping import run as run_create_mapping
 from dcicutils.log_utils import set_logging
-from dcicutils.beanstalk_utils import whodaman
-from dcicutils.env_utils import is_test_env, is_hotseat_env, guess_mirror_env, is_stg_or_prd_env
+from dcicutils.beanstalk_utils import compute_ff_prd_env, compute_cgap_prd_env
```
Comment (j1z0) [inline src/encoded/commands/create_mapping_on_deploy.py:8]: Codacy Issue found: [Unused compute_ff_prd_env imported from dcicutils.beanstalk_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

File: src/encoded/commands/create_mapping_on_deploy.py [hunk 2]
```diff
@@ -5,8 +5,10 @@
...
+from dcicutils.deployment_utils import DeployConfigManager
+from dcicutils.env_utils import is_test_env, is_hotseat_env, guess_mirror_env, is_stg_or_prd_env, is_fourfront_env
```
Comment (j1z0) [inline src/encoded/commands/create_mapping_on_deploy.py:10]: Codacy Issue found: [Unused is_test_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240)

**Comment by netsettler** [comment]
I guess there are two different DRAFT mechanisms. I just wanted the one that the PR itself seems to understand. Anyway, this will remain a draft until the utils change is firm.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         

Clones added
============
- src/encoded/tests/test_create_mapping.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

File: src/encoded/tests/test_create_mapping.py [hunk 3]
```diff
@@ -1,8 +1,14 @@
 import pytest
 
+from dcicutils.deployment_utils import CreateMappingOnDeployManager
+from dcicutils.misc_utils import ignored
```
Comment (j1z0) [inline src/encoded/tests/test_create_mapping.py:4]: Codacy Issue found: [Unused ignored imported from dcicutils.misc_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         

Clones added
============
- src/encoded/tests/test_create_mapping.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         

Clones added
============
- src/encoded/tests/test_create_mapping.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         

Clones added
============
- src/encoded/tests/test_create_mapping.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

**Comment by netsettler** [comment]
@willronchetti, if you want to take this back and continue working on it, we can arrange to have it merge into your branch. If you were done with things and want to merge to master, we can do that. But please either way thoroughly review that the new unit tests are showing the behavior you want. I hope I've organized them where you can see the various quantities you'd want to be testing. If you need adjustments, I can do that or you should feel free as well.

File: src/encoded/commands/create_mapping_on_deploy.py [hunk 4]
```diff
@@ -138,76 +138,53 @@ def get_my_env(app):
...
-    return deploy_cfg
-
-
 def _run_create_mapping(app, args):
     """
     Runs create_mapping with deploy options and report errors. Allows args passed from argparse in main to override
-    the default deployment configuration
+    the default deployment configuration.
+
+    XXX: Should be refactored into CMDeployer in dcicutils.deployment_utils.py - Will 5/22/2020
```
Comment (willronchetti) [inline src/encoded/commands/create_mapping_on_deploy.py:146]: This comment can be removed now 😄

File: src/encoded/tests/test_create_mapping.py [hunk 5]
```diff
@@ -55,87 +57,220 @@ def test_create_mapping_item_order(registry):
...
+@patch('dcicutils.deployment_utils.compute_ff_prd_env', MagicMock(return_value='fourfront-green'))
+@patch('encoded.commands.create_mapping_on_deploy.get_my_env', MagicMock(return_value='fourfront-hotseat'))
+def test_get_deployment_config_hotseat():
+    """ Tests get_deployment_config in the hotseat case with a new-style ecosystem. """
+    my_env = create_mapping_on_deploy.get_my_env('ignored-for-mock')
+    assert my_env == 'fourfront-hotseat'
+    cfg = CreateMappingOnDeployManager.get_deploy_config(env=my_env, args=MockedCommandArgs(), log=MockedLog())
+    assert cfg['ENV_NAME'] == my_env  # sanity
+    assert cfg['SKIP'] is True  # The other values don't matter if this is set.
+    # assert cfg['WIPE_ES'] is ...
```
Comment (willronchetti) [inline src/encoded/tests/test_create_mapping.py:116]: No need to check these since they are not relevant.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/encoded/commands/create_mapping_on_deploy.py  -3
         

Clones added
============
- src/encoded/tests/test_create_mapping.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5573240&bid=18273092)

File: src/encoded/commands/create_mapping_on_deploy.py (hunk 4 above)
Comment (netsettler) [reply src/encoded/commands/create_mapping_on_deploy.py]: Done.

File: src/encoded/tests/test_create_mapping.py (hunk 5 above)
Comment (netsettler) [reply src/encoded/tests/test_create_mapping.py]: Comments trimmed.

**Pull request merged**